### PR TITLE
An example program to make JSON Post request with HttpURLConnection

### PR DESCRIPTION
**Pull request by cupendra**
@pivovarit I created a new pull request for BAEL-2737 (HttpUrlConnection POST Request with JSON). Please review the same

### CODE REVIEW

File: core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+				String responseLine = null;
+				while ((responseLine = br.readLine()) != null) {
+					response.append(responseLine.trim());
+				}
+				System.out.println(response.toString());
+			}
+		}catch (IOException ex) {
+			ex.printStackTrace();
+		}finally {
+			con = null;
```
Comment (pivovarit) [inline core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java]: Not sure why to null it out at the end, it's the end of the main() anyway.

File: core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class PostJSONWithHttpURLConnection {
+	
+	public static void main (String []args) {
+		HttpURLConnection con = null;
```
Comment (pivovarit) [inline core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java]: No need to keep that outside the try-catch I think

File: core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+			try(BufferedReader br = new BufferedReader(new InputStreamReader(con.getInputStream(), "utf-8"))){
+				StringBuilder response = new StringBuilder();
+				String responseLine = null;
+				while ((responseLine = br.readLine()) != null) {
+					response.append(responseLine.trim());
+				}
+				System.out.println(response.toString());
+			}
+		}catch (IOException ex) {
+			ex.printStackTrace();
```
Comment (pivovarit) [inline core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java]: Would be good to rethrow unchecked exception and not just swallow it

Comment (cupendra) [reply core-java/src/main/java/com/baeldung/urlconnection/PostJSONWithHttpURLConnection.java]: Hi, @pivovarit I changed the code as suggested. Please take look it now.

**Pull request merged**